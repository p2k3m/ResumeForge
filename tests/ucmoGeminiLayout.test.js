import { jest } from '@jest/globals';

const setContent = jest.fn();
const pdf = jest.fn().mockResolvedValue(Buffer.from('PDF'));
const close = jest.fn();
const newPage = jest.fn().mockResolvedValue({ setContent, pdf });
const launch = jest.fn().mockResolvedValue({ newPage, close });

jest.unstable_mockModule('puppeteer', () => ({ default: { launch } }));

const { generatePdf } = await import('../server.js');

test('ucmo template uses Gemini layout markup', async () => {
  const markup = '<html><body><table id="top"><tr><td>contact</td><td><img src="logo.png"/></td></tr></table></body></html>';
  const generativeModel = {
    generateContent: jest
      .fn()
      .mockResolvedValue({ response: { text: () => markup } })
  };
  await generatePdf('Jane Doe\nExperience', 'ucmo', {}, generativeModel);
  expect(generativeModel.generateContent).toHaveBeenCalled();
  const prompt = generativeModel.generateContent.mock.calls[0][0];
  expect(prompt).toMatch(/University of Central Missouri/);
  expect(setContent).toHaveBeenCalledWith(markup, { waitUntil: 'networkidle0' });
  expect(setContent.mock.calls[0][0]).toMatchSnapshot();
});
