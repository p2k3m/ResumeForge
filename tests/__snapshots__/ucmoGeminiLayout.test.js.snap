// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ucmo template uses Gemini layout markup 1`] = `"<html><body><table id="top"><tr><td>contact</td><td><img src="logo.png"/></td></tr></table></body></html>"`;
